sympy/functions/combinatorial/numbers.py

```diff
@@ -1858,12 +1858,50 @@ def stirling(n, k, d=None, kind=2, signed=False):
 
 
 @cacheit
-def _nT(n, k):
+def _nT(*args):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if len(args) == 2:
+        n = args[0]
+        k = args[1]
+        if k > n:
+            return 0
+        if k >= n/2.0:
+            args = [n - k]
+    numparts = 0
+    if len(args) == 1:
+        n = args[0]
```

Comment:
This is essentially the same algorithm as the one implemented in `partition`:
https://github.com/sympy/sympy/blob/4cdcbd914d66733eee1de77d592bbad3d693b049/sympy/functions/combinatorial/numbers.py#L1351-L1366

Would it be possible to use that here?

Reply:
Hey please tell if this will work ,I will replace everything with 
numparts = partition(n)
when len(args) == 1

Reply:
What is `k` when there is 1 arg? There is no need to remove the args that are there. You can choose to handle the different cases for k and n as you see fit, however.